spec/codecs/cef_spec.rb

```diff
@@ -306,32 +349,45 @@ def validate(e)
       subject(:codec) { LogStash::Codecs::CEF.new("delimiter" => '\r\n') }
 
       it "should parse on the delimiter " do
-        subject.decode(message) do |e|
+        do_decode(subject,message) do |e|
           raise Exception.new("Should not get here. If we do, it means the decoder emitted an event before the delimiter was seen?")
         end
 
-        event = false;
-        subject.decode("\r\n") do |e|
+        decode_one(subject, "\r\n") do |e|
           validate(e)
           insist { e.get("deviceVendor") } == "security"
           insist { e.get("deviceProduct") } == "threatmanager"
-          event = true
         end
+      end
+    end
+
+    context 'when a CEF header ends with a pair of properly-escaped backslashes' do
+      let(:backslash) { '\\' }
+      let(:pipe) { '|' }
+      let(:message) { "CEF:0|security|threatmanager|1.0|100|double backslash" +
+                      backslash + backslash + # escaped backslash
+                      backslash + backslash + # escaped backslash
+                      "|10|src=10.0.0.192 dst=12.121.122.82 spt=1232" }
 
-        expect(event).to be_truthy
+      it 'should include the backslashes unescaped' do
+        event = decode_one(subject, message)
+
+        expect(event).to be_a(LogStash::Event)
```

Comment:
minor: I think this expectation is superfluous for this test. I suggest a separate test be made just to test that the decode function produces an `Event` class.

Reply:
The existing `validate` function that is called on most of the rest of the specs does this before testing the contents of the event, so I was mostly copying for consistency.

You're right, though. It's not strictly necessary.